### PR TITLE
Remove "U" modifier from regexes

### LINKED ISSUE #267: Empty result for big reports with writeHTML()
The error was noticed 10 years ago. When trying to generate a PDF from a large HTML file, an empty file is generated, and no error messages are generated.

The report generation dies because of a preg_replace() used by TCPDF. It was a PHP bug for older versions (mine is 5.3.3), preg_replace() returns null silently when the string is too long. See this:

https://sourceforge.net/p/tcpdf/discussion/435311/thread/f7f96b04/
https://stackoverflow.com/questions/14870397/tcpdf-displays-blank-pdf-if-query-result-is-big

### PR DESCRIPTION
**Pull request by mvorisek**
`U` modifier invert greediness which is hard to read, thus remove it

### CODE REVIEW

File: tcpdf.php [hunk 1]
```diff
@@ -16350,15 +16350,15 @@ protected function getHtmlDomArray($html) {
 		$css = array();
 		// get CSS array defined at previous call
 		$matches = array();
-		if (preg_match_all('/<cssarray>([^\<]*)<\/cssarray>/isU', $html, $matches) > 0) {
+		if (preg_match_all('/<cssarray>([^\<]*?)<\/cssarray>/is', $html, $matches) > 0) {
 			if (isset($matches[1][0])) {
 				$css = array_merge($css, json_decode($this->unhtmlentities($matches[1][0]), true));
 			}
-			$html = preg_replace('/<cssarray>(.*?)<\/cssarray>/isU', '', $html);
+			$html = preg_replace('/<cssarray>(.*?)<\/cssarray>/is', '', $html);
```
Comment (mvorisek) [inline tcpdf.php:16357]: previous behaviour was without the `?`, correct now?

File: tcpdf.php [hunk 2]
```diff
@@ -16396,8 +16396,8 @@ protected function getHtmlDomArray($html) {
 		// create a special tag to contain the CSS array (used for table content)
 		$csstagarray = '<cssarray>'.htmlentities(json_encode($css)).'</cssarray>';
 		// remove head and style blocks
-		$html = preg_replace('/<head([^\>]*)>(.*)?<\/head>/siU', '', $html);
-		$html = preg_replace('/<style([^\>]*)>([^\<]*)<\/style>/isU', '', $html);
+		$html = preg_replace('/<head([^\>]*?)>(.*)<\/head>/is', '', $html);
```
Comment (mvorisek) [inline tcpdf.php:16399]: I belive this will bring back the #267 issue

@eisberg please test

Comment (eisberg) [reply tcpdf.php:16399]: Yes, I tested, with this option, an empty file is generated. This bring back the #267 issue

```
$html = preg_replace('/<head([^\>]*)>(.*)?<\/head>/siU', '', $html); // Works correctly
```

```
$html = preg_replace('/<style([^\>]*?)>([^\<]*?)<\/style>/is', '', $html); // It doesn't work properly, the content is deleted
```

Comment (mvorisek) [inline tcpdf.php:16399]:
```suggestion
		$html = preg_replace('/<head([^\>]*?)>(.*?)<\/head>/is', '', $html);
```
this works, right? Please also provide a link to full test file.

Comment (eisberg) [reply tcpdf.php:16399]: yes, this works

example mini test project:  [test-tcpdf.zip](https://github.com/tecnickcom/TCPDF/files/6271174/test-tcpdf.zip)
Unpack, then run:

```
composer up
php ./test.php
```

Comment (mvorisek) [reply tcpdf.php]: resolved in https://github.com/tecnickcom/TCPDF/pull/359#discussion_r608555448

**Comment by nicolaasuni** [comment]
This change seems risky and it is not necessary. I would like to see a test for all cases.

**Comment by mvorisek** [comment]
> 
> 
> This change seems risky and it is not necessary. I would like to see a test for all cases.

`U` modifier is hard to read as it inverts the greediness/lazyness of the quantifiers. This PR removed usage of `U` regex modifier. The function remains completely unchanged. Feel free to test this statement on https://3v4l.org/ website.

**Review by nicolaasuni — APPROVED** [review]

**Pull request merged**